hyperspy/signal.py

```diff
@@ -5220,6 +5222,9 @@ def map(
             See docstring for output_signal_size for more information.
             Default None.
         %s
+        warn_non_uniform_axis : bool
```

Comment:
Would it be better to have a generic `silence_warnings` rather than adding this for every different warning? 

so:

```python
silence_warnings =True # all warnings silenced
silence_warnings =False # default no warnings silenced
silence_warnings = ["non-uniform", "scales"] # silence both warnings and leaves oppertunity for more warnings...
```

Reply:
Done! 😺